js/http.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+import { Closer } from "./io";
+import * as msg from "gen/msg_generated";
+import { assert, log } from "./util";
+import * as dispatch from "./dispatch";
+import * as flatbuffers from "./flatbuffers";
+import { close } from "./files";
+
+// TODO Cannot use Headers due to bug in ts_declaration_builder.
+// import { Headers } from "./headers";
+// import * as headers from "./headers";
+// import * as domTypes from "./dom_types";
+
+type HttpHandler = (req: ServerRequest, res: ServerResponse) => void;
+
+export class HttpServer implements Closer {
```

Comment:
Crazy ass idea, but it seems like this could lend itself to async generators... if we implemented an async iterator in here as well, we could do:

```ts
for await (const [req, res] of deno.httpListen(addr)) {
  // ... fun stuff
}
```

Which would continue to loop until closed.  The generator would look something like this:

```ts
[Symbol.asyncIterator]: async function* () {
  while (this.closing === false) {
    yield await httpAccept(this.rid);
  }
}
```

Reply:
Seems like a cute idea. I'm open to it. I think it should be something like

```
let server = deno.httpServe(addr);
for await (const [req, res] of server.iter()) {
   / ...
}
```